drivers/peci/peci_ite_it8xxx2.c

```diff
@@ -0,0 +1,355 @@
+
+static void peci_it8xxx2_init_vtts(struct peci_it8xxx2_regs *reg_base,
+					enum peci_vtts vol_opt)
+{
+	reg_base->PADCTLR = (reg_base->PADCTLR & PECI_DVIE) | vol_opt;
+}
+
+static void peci_it8xxx2_rst_status(struct peci_it8xxx2_regs *reg_base)
+{
+	reg_base->HOSTAR = HOSTAR_RST_ANYBIT;
+}
+
+static int peci_it8xxx2_check_host_busy(struct peci_it8xxx2_regs *reg_base)
+{
+	return (reg_base->HOSTAR & HOBY) ? (-EBUSY) : 0;
+}
+
+static int peci_it8xxx2_check_host_finish(const struct device *dev)
+{
+	struct peci_it8xxx2_data *data = dev->data;
+	const struct peci_it8xxx2_config *config = dev->config;
+	struct peci_it8xxx2_regs *const peci_regs =
+		(struct peci_it8xxx2_regs *)config->base_addr;
+
+	int ret = k_sem_take(&data->device_sync_sem, K_MSEC(PECI_TIMEOUT_MS));
+
+	if (ret == -EAGAIN) {
+		LOG_ERR("%s: Timeout", __func__);
+		return -ETIMEDOUT;
+	}
+
+	if (peci_regs->HOSTAR != FINISH) {
+		LOG_ERR("[PECI] Error: HOSTAR=0x%02X\r\n", peci_regs->HOSTAR);
+			return -EIO;
+	}
+
+	return 0;
+}
+
+static int peci_it8xxx2_configure(const struct device *dev, uint32_t bitrate)
+{
+	struct peci_it8xxx2_data *data = dev->data;
+	const struct peci_it8xxx2_config *config = dev->config;
+	struct peci_it8xxx2_regs *const peci_regs =
+		(struct peci_it8xxx2_regs *)config->base_addr;
+
+	uint8_t hoctl2r_to_write;
+
+	data->bitrate =  bitrate;
+
+	hoctl2r_to_write =
+		(peci_regs->HOCTL2R) & (~(PECI_IT8XXX2_BITRATE_BITS_MASK));
+
+	switch (bitrate) {
+	case PECI_IT8XXX2_BITRATE_2MHZ:
+		break;
+
+	case PECI_IT8XXX2_BITRATE_1MHZ:
+		hoctl2r_to_write |= PECI_IT8XXX2_BITRATE_1MHZ_BITS;
+		break;
+
+	case PECI_IT8XXX2_BITRATE_1P6MHZ:
+		hoctl2r_to_write |= PECI_IT8XXX2_BITRATE_1P6MHZ_BITS;
+		break;
+
+	default:
+		LOG_ERR("[PECI] Error: Specified Bitrate Not Supported\r\n");
+		hoctl2r_to_write |= PECI_IT8XXX2_BITRATE_1MHZ_BITS;
+		data->bitrate =  PECI_IT8XXX2_BITRATE_1MHZ;
+		peci_regs->HOCTL2R = hoctl2r_to_write;
+		return -ENOTSUP;
+	}
+
+	peci_regs->HOCTL2R = hoctl2r_to_write;
+
+	return 0;
+}
+
+static int peci_it8xxx2_enable(const struct device *dev)
+{
+	const struct peci_it8xxx2_config *config = dev->config;
+	struct peci_it8xxx2_regs *const peci_regs =
+		(struct peci_it8xxx2_regs *)config->base_addr;
+
+	peci_regs->HOCTLR |= (FIFOCLR|FCSERR_ABT|PECIHEN|CONTROL);
+
+	return 0;
+}
+
+static int peci_it8xxx2_disable(const struct device *dev)
+{
+	const struct peci_it8xxx2_config *config = dev->config;
+	struct peci_it8xxx2_regs *const peci_regs =
+		(struct peci_it8xxx2_regs *)config->base_addr;
+
+	peci_regs->HOCTLR &= ~(PECIHEN);
+	return 0;
+}
+
+static void peci_it8xxx2_rst_module(const struct device *dev)
+{
+	const struct peci_it8xxx2_config *config = dev->config;
+	struct peci_it8xxx2_regs *const peci_regs =
+		(struct peci_it8xxx2_regs *)config->base_addr;
+
+	peci_regs->HOSTAR = HOSTAR_RST_ANYBIT;
+
+	LOG_ERR("[PECI] Module Reset for Status Error.\r\n");
+	/* Reset IT8XXX2 PECI Module Thoroughly */
+	IT83XX_GCTRL_RSTC4 |= RPECI;
```

Comment:
The Chromium EC driver included a [delay](https://chromium.googlesource.com/chromiumos/platform/ec/+/01aa26ee/chip/it83xx/peci.c#66) after this reset.  Is that still required?

Reply:
So far I run the test code with the IT81302, and it seems working without putting a delay. I'll make more experiments on this part.

Reply:
So far we've run it for days, it looks good without the delay after resets.